evap/staff/forms.py

```diff
@@ -227,6 +228,9 @@ def __init__(self, *args, **kwargs):
         if self.instance.pk and not self.instance.can_manager_edit:
             disable_all_fields(self)
 
+        if self.instance.pk:
+            self.instance.old_course = self.instance.course
```

Comment:
Attaching additional information like this works, the attached information will still be there when the form is constructed again from request.POST? I'd be really surprised.

Reply:
i'm actually a bit confused, but the `if hasattr(self.instance, 'old_course'):` in line 269 is doing stuff when the course changed so yeah. magic.
this was influenced by https://stackoverflow.com/questions/23361057/django-comparing-old-and-new-field-value-before-saving

Reply:
Impressive. Using [changed_data](https://docs.djangoproject.com/en/2.1/ref/forms/api/#checking-which-form-data-has-changed) was not possible here? If it was, I would really prefer that.

Reply:
changed_data returns only the name of fields that changed. i need to know what the old value was.

Reply:
Hmm make sense. In this case, I'd suggest to add a test that this works, then I'm fine with it (or do we already have one? I just quickly checked and did not find anything)

Reply:
I didn't add any tests yet